doc/embedsource.md

```diff
@@ -0,0 +1,69 @@
+# Using picolibc as an embedded source
+
+For projects where deterministic builds, reproduceability and traceability is important, it is possible to embed the entire picolibc library as a meson subproject into the source directory of the master project. The benefit is that git submodules and meson subprojects work nicely together, and it is possible to lock down the libc version together with the source tree.
+
+For more information on the meson subproject feature, refer to the documentation at : https://mesonbuild.com/Subprojects.html
+
+## Adding picolibc as a subproject
+
+First you need to add picolibc as a git submodule:
+
+  ```
+  mkdir subprojects
+  git submodule add https://github.com/keith-packard/picolibc subprojects/picolibc
+  ```
+
+Your source direcoty could now look something like this:
+
+  ```
+  subprojects/
+    picolibc/
+    otherlib/
+  src/
+    main.c
+  meson.build
+  ```
+
+## Add dependency to meson.build
+
+First of all, when using picolibc as an embedded source, you should not specify -specs=picolibc.specs in the c_args.
```

Comment:
Hrm. If you created a custom .specs file in the embedded directory, you could use that. This would also solve the problem of wanting to avoid any installed libc as the .specs file would override that path.